src/plugins/unified_doc_viewer/public/components/doc_viewer_table/table.tsx

```diff
@@ -493,6 +508,23 @@ export const DocViewerTable = ({
           <EuiFlexItem grow={false}>
             <EuiSpacer size="s" />
           </EuiFlexItem>
+          <EuiFlexItem
+            grow={false}
+            css={css`
+              align-self: end;
+              padding-bottom: ${euiTheme.size.s};
+            `}
+          >
+            <EuiSwitch
+              label={i18n.translate('unifiedDocViewer.hideNullValues.switchLabel', {
+                defaultMessage: 'Hide fields with null values',
+              })}
+              checked={areNullValuesHidden ?? false}
+              onChange={onHideNullValuesChange}
+              compressed
+              data-test-subj="unifiedDocViewerHideNullValuesSwitch"
+            />
+          </EuiFlexItem>
```

Comment:
I think we should not show "Hide fields with null values" switch in data view mode. Null values are not present by default there. And not all documents have all fields in source. Toggling the switch would not bring them back anyway. So it would be confusing to have the switch if it can only affect some meta fields filtering.

<img width="1698" alt="Screenshot 2024-08-01 at 10 08 24" src="https://github.com/user-attachments/assets/67e77eef-7d96-4a22-9431-0f5d26389e1f">

For ES|QL I agree it makes sense to allow toggling with the switch since records always have all columns present.

<img width="1418" alt="Screenshot 2024-08-01 at 10 09 31" src="https://github.com/user-attachments/assets/6bb199bf-e622-4690-ae6d-94375b151a9f">

Reply:
Done https://github.com/elastic/kibana/pull/189601/commits/92b4f8ddd3eb9f8363719353260f491fe21eccdc I also initialized the flaky test runner again https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6662